Libs/MRML/Logic/vtkMRMLApplicationLogic.h

```diff
@@ -316,6 +327,8 @@ class VTK_MRML_LOGIC_EXPORT vtkMRMLApplicationLogic
   class vtkInternal;
   vtkInternal* Internal;
 
+  static vtkMRMLTranslator* TranslatorInstance;
```

Comment:
When is this going to be deleted?
How can you ensure that it is deleted at the right time (not too early - when something may still call it; but not too late - after the VTK leak detection mechanism already counted all the non-deleted VTK objects)?

How would we implement this correctly on Windows, Linux, and macOS? We should not reinvent this from scratch but follow a singleton pattern already used in Slicer or VTK.

Reply:
In this particular case, I don't think we need to use [vtkSingleton.h](https://github.com/Slicer/Slicer/blob/main/Libs/MRML/DisplayableManager/vtkSingleton.h) help class.

Since the lifecycle of this class is driven by `vtkMRMLApplicationLogic` (itself instantiated and deleted by the main application), the reference to `vtkMRMLTranslator` will be released upon destruction of `vtkMRMLApplicationLogic`.

Reply:
+1 for making a simple vtk class that is owned by the application logic.